python/lsst/ctrl/mpexec/preExecInit.py

```diff
@@ -0,0 +1,183 @@
+# This file is part of ctrl_mpexec.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ['PreExecInit']
+
+# -------------------------------
+#  Imports of standard modules --
+# -------------------------------
+import logging
+
+# -----------------------------
+#  Imports for other modules --
+# -----------------------------
+
+_LOG = logging.getLogger(__name__.partition(".")[2])
+
+
+class PreExecInit:
+    """Initialization of registry for QuantumGraph execution.
+
+    This class encapsulates all necessary operations that have to be performed
+    on butler and registry to prepare them for QuanrumGraph execution.
+
+    Parameters
+    ----------
+    butler : `~lsst.daf.butler.Butler`
+        Data butler instance.
+    """
+    def __init__(self, butler):
+        self.butler = butler
```

Comment:
Given that it would be nice to run at least some of these operations on a `Pipeline` in the future as well as a `QuantumGraph`, what do you think about passing the `QuantumGraph` (either instead of or in addition to the `Butler`) at construction instead of to each of the methods?  That would let us have a different class in the future with (mostly) the same method interfaces to handle `Pipelines`.

Reply:
Hmm, some of that interface cannot even work with Pipeline, so I'm not sure how we can make it work. We could probably define some polymorphic interface for that but if it's going to work differently for graph and pipeline it will not be truly polymorphic. I'd say that having separate interface for Pipeline (maybe even in this same class) is probably better and less confusing.